### PR TITLE
refactor ChainListener::processMessage

### PR DESCRIPTION
**Pull request by seelabs**

### CODE REVIEW

**Review by pwang200 — APPROVED** [review]
LGTM. (I didn't test.)

File: src/xbwd/client/ChainListener.cpp [hunk 1]
```diff
@@ -348,15 +297,70 @@ ChainListener::processMessage(Json::Value const& msg)
...
+
+        if (!fieldMatchesStr(msg, ripple::jss::type, ripple::jss::transaction))
+            return {};
+
+        if (!msg.isMember(ripple::jss::transaction))
+            return {};
+
         auto const txn = msg[ripple::jss::transaction];
+
+        static std::array<std::pair<TxnType, ::Json::StaticString>, 7> const
```
Comment (oleks-rip) [inline src/xbwd/client/ChainListener.cpp:330]: why not std::unordered_map<std::string,TxnType > ?

File: src/xbwd/client/ChainListener.cpp [hunk 2]
```diff
@@ -367,6 +371,10 @@ ChainListener::processMessage(Json::Value const& msg)
             pushEvent(std::move(e));
             return;
         }
+        else
+        {
```
Comment (oleks-rip) [inline src/xbwd/client/ChainListener.cpp:375]: Maybe some log here

Comment (seelabs) [reply src/xbwd/client/ChainListener.cpp:384]: Fixed in https://github.com/seelabs/xbridge_witness/pull/13/commits/199820ed53f0f4f856c99dc8153cd8016156346e

File: src/xbwd/client/ChainListener.cpp (hunk 1 above)
Comment (seelabs) [reply src/xbwd/client/ChainListener.cpp]: Good idea. Fixed in https://github.com/seelabs/xbridge_witness/pull/13/commits/199820ed53f0f4f856c99dc8153cd8016156346e

**Review by oleks-rip — APPROVED** [review]
LGTM

**Comment by seelabs** [comment]
merged

**Pull request closed without merging**